packages/editor/src/components/rich-text/index.js

```diff
@@ -437,8 +437,10 @@ export class RichText extends Component {
 		const { keyCode } = event;
 
 		if (
-			( keyCode === BACKSPACE && isHorizontalEdge( rootNode, true ) ) ||
-			( keyCode === DELETE && isHorizontalEdge( rootNode, false ) )
+			window.getSelection().isCollapsed && (
+				( keyCode === BACKSPACE && isHorizontalEdge( rootNode, true ) ) ||
```

Comment:
For some reason I would have thought that `isHorizontalEdge` also does the check, as navigating also requires it.

Reply:
Yes, I noticed it was also checked in the `WritingFlow`. I didn't want to change `isHorizontalEdge` behavior as I wasn't sure about the implications

Reply:
Maybe. As it reads, I see no reason why `isHorizontalEdge` should have any concern about whether the selection is collapsed or not. This assumes there's a use case for wanting to know if there's an uncollapsed selection at the horizontal edge. We don't have this need in our project, which might bias our opinion on it. Or maybe "is horizontal edge" is meant to mean that pressing ArrowLeft/ArrowRight would have no effect (in a traditional input).